### PR TITLE
Replace inlined secrets with environment variable

### LINKED ISSUE #525: Replace inline secret with environment variable name
Instead of:

```yaml
- github_app: <token>
  github_organization: change-metrics
```

... use the environment to define secret:

```yaml
- github_app: MY_GITHUB_APP_NAME
  ...
```

### PR DESCRIPTION
**Pull request by TristanCacqueray**
Fixes #525

### CODE REVIEW

**Comment by TristanCacqueray** [comment]
@morucci that should be good for review, and it seems like we could use some cursory tests for the macroscope functions.

File: README.md [hunk 1]
```diff
@@ -327,22 +317,29 @@ the initial date.
 
 ### Full configuration file example
 
+Here are the expected environment variables:
+
+- `CRAWLERS_API_KEY`: an arbitrary api key used by the crawler to index data.
+- `GITHUB_TOKEN`: a github token for `github_organization` crawler.
+- `GITLAB_TOKEN`: a gitlab token for `gitlab_organization` crawler.
+- Optional `GERRIT_PASSWORD` for `gerrit_url` crawler.
```
Comment (morucci) [inline README.md:325]: We should use GitHub, GitLab, Gerrit name when we talk about crawlers in the doc.

File: README.md [hunk 2]
```diff
@@ -311,7 +301,7 @@ Check the OpenAPI definitions for tasks data endpoints: [Monocle OpenAPI][monocl
 ```YAML
 workspaces:
   - name: default
-    crawlers_api_key: 1a2b3c4d5e
+    crawlers_api_key: API_KEY_ENV
```
Comment (morucci) [inline README.md:304]: So I guess we can omit this crawlers_api_key as it is optional and then the backend is looking for CRAWLERS_API_KEY env var.

File: haskell/src/Monocle/Api/Config.hs [hunk 3]
```diff
@@ -182,17 +189,19 @@ getAliases index = maybe [] (fmap toTuple) (search_aliases index)
 
 getCrawlerProject :: Crawler -> [Text]
 getCrawlerProject Crawler {..} = case provider of
-  GitlabProvider Gitlab {..} -> fromMaybe [] gitlab_repositories
+  GitlabProvider Gitlab {..} ->
+    let addOrgPrefix repo = gitlab_organization <> "/" <> repo
```
Comment (morucci) [inline haskell/src/Monocle/Api/Config.hs:193]: we should trim the trailing '/' for the organization name.

**Review by morucci — COMMENTED** [review]
I'll give it a try locally.

**Comment by morucci** [comment]
It seems we need to update the CONTRIBUTING.rst doc to reflect the need to have export secrets before running the apis and crawlers.

File: haskell/src/Monocle/Api/Server.hs [hunk 4]
```diff
@@ -167,7 +167,7 @@ crawlerAddDoc request = do
             `orDie` CrawlerPB.AddDocErrorAddUnknownCrawler
 
         when
-          (Config.crawlers_api_key index /= toStrict apiKey)
+          (Config.crawlers_api_key index /= Just (toStrict apiKey))
```
Comment (morucci) [inline haskell/src/Monocle/Api/Server.hs:170]: Don't we need to read from the environment ?

Comment (morucci) [reply haskell/src/Monocle/Api/Server.hs]: I've attempted to provide a fix for this.

**Comment by morucci** [comment]
Ok after my change I was able to index some gitlab and github repos.
I haven't tested the docker deployment btw.

File: contrib/start-crawlers-legacy.sh [hunk 5]
```diff
@@ -0,0 +1,15 @@
...
+    python3 -mvenv .venv
+    ./.venv/bin/pip install pip
+    ./.venv/bin/pip install -r requirements.txt
+fi
+
+if ! test -f .venv/bin/monocle; then
+    ./.venv/bin/python3 setup.py install
+fi
+
+exec monocle --elastic-conn localhost:19200 crawler --config etc/config.yaml
```
Comment (morucci) [inline contrib/start-crawlers-legacy.sh:15]: Ah crap I need to set the port to 9200 by default

**Review by morucci — APPROVED** [review]

**Pull request merged**